Cargo.lock

```diff
@@ -399,6 +400,15 @@ dependencies = [
  "winapi 0.2.8 (registry+https://github.com/rust-lang/crates.io-index)",
 ]
 
+[[package]]
+name = "mio-uds"
```

Comment:
My understanding is that we only need to set a flag on the first SIGTERM or SIGINT (or equivalents, optionally, on Windows) sent to the main thread, to shut down the main thread's event loop. That is the signal is sent to the same thread that the event loop is running on, IIUC. In that case, isn't tokio-signal overkill, since it is designed to support handling arbitrary signals on arbitrary event loops? In particular, tokio-signal creates the UDS to allow this cross-thread processing that we don't need, right? And also it uses a `Stream` to support a signal being handled multiple times but we only need to handle it once.

Is this your understanding too? If so, we should consider, in the future, replacing this dependency with a simpler one.

Reply:
Yes, much of that is true. This also factored in to my decision:

- There's a multitude of warnings about how badly signal handler functions can mess up programs, with a list of things NOT to do within them. In that case, I'd rather use at least some library where people have used care to not trigger those problems. It could be `tokio-signal`, or something much simpler, maybe `signal`.
- Having a `Future` works great with the existing system we already have in place, so `tokio-signal` providing that for us is icing.
- We do only need a `Future`, not a `Stream`, at least for right now. Though, the overhead there is minimal. And it's possible that in we may want to handle receiving multiple signals at some point, such as perhaps the second signal could imply "hurry up with the shutdown", or similar.
- Knowing that this is safe and plugs in easily gives ease-of-mind, and the overhead of receiving signals shouldn't affect the proxy, as we don't expect to be receiving and processing signals on an extremely frequent basis.

I'm not for wasting unnecessarily, and would welcome a library that perhaps did less, while still being safe and convenient to use. Probably not high priority, though. Does that all sound reasonable?

Reply:
> Probably not high priority, though. Does that all sound reasonable?

Yes, that all sounds reasonable to me.